source/windows/secure_channel_tls_handler.c

```diff
@@ -1222,14 +1222,21 @@ static int s_process_pending_output_messages(struct aws_channel_handler *handler
             sc_handler->buffered_read_out_data_buf.len = 0;
         }
     }
-    if (sc_handler->buffered_read_out_data_buf.len == 0 && sc_handler->delay_shutdown_scheduled) {
-        sc_handler->read_shutdown_completed = true;
+    ret = AWS_OP_SUCCESS;
+    if (sc_handler->buffered_read_out_data_buf.len > 0) {
+        /* Still have more data to be delivered */
+        return ret;
+    }
+
+done:
+    if (sc_handler->read_state == AWS_TLS_HANDLER_READ_SHUTTING_DOWN) {
+        sc_handler->read_state = AWS_TLS_HANDLER_READ_SHUT_DOWN_COMPLETE;
         /* Continue the shutdown process delayed before. */
         aws_channel_slot_on_handler_shutdown_complete(
-            sc_handler->slot, AWS_CHANNEL_DIR_READ, sc_handler->delay_shutdown_error_code, false);
+            sc_handler->slot, AWS_CHANNEL_DIR_READ, sc_handler->shutdown_error_code, false);

```

Comment:
if a local error happens, and `sc_handler->shutdown_error_code == 0`, we should pass along the local error code instead

maybe do a local `int shutdown_error_code = 0;` like in s2n/darwin